android/capacitor/src/main/java/com/getcapacitor/Bridge.java

```diff
@@ -614,6 +627,38 @@ public void releaseCall(PluginCall call) {
         this.savedCalls.remove(call.getCallbackId());
     }
 
+    /**
+     * Removes the earliest saved call prior to a permissions request for a given plugin and
+     * returns it.
+     *
+     * @return The saved plugin call
+     */
+    protected PluginCall getPermissionCall(String pluginId) {
+        LinkedList<String> permissionCallIds = this.savedPermissionCallIds.get(pluginId);
+        String savedCallId = null;
+        if (permissionCallIds != null) {
+            savedCallId = permissionCallIds.poll();
+        }
+
+        return getSavedCall(savedCallId);
+    }
+
+    /**
+     * Save a call to be retrieved after requesting permissions. Calls are saved in order.
+     *
+     * @param call The plugin call to save.
+     */
+    public void savePermissionCall(PluginCall call) {
```

Comment:
Does this need to be public?

Reply:
Not any more, I can also refine the permission requesting helpers on Plugin to save the call so the plugin author won't need to call this at all